### PR TITLE
chgres_cube: Add a check at static data read for incompatible old static data

### PR DESCRIPTION
**Pull request by LarissaReames**
## DESCRIPTION OF CHANGES: 
chgres_cube would produce incorrect surface data or fail with an unhelpful message in debug mode when used with old static data after PR #873 . I added a check in the read of the static fields that looks for any points where landfrac > 0 but static data = -999 (missing data). The program now fails at this point when used with incompatible static data and provides a useful error message directing the user to create new static data with an up-to-date UFS_UTILS build. 

## TESTS CONDUCTED: 
If there are changes to the build or source code, the tests below must be conducted. Contact a repository manager if you need assistance.

- [x] Compile branch on all Tier 1 machines using Intel (Orion, Jet, Hera, Hercules and WCOSS2). Done using 8608666. 
- [X] Compile branch on Hera using GNU. Done using 8608666.
- [x] Compile branch in 'Debug' mode on WCOSS2. Done using 8608666.
- [X] Run unit tests locally on any Tier 1 machine. Done on Jet using 8608666.
- [x] Run relevant consistency tests locally on all Tier 1 machine. The chgres_cube regression tests were tested using 8608666. All tests passed as expected.

Also tested with RRFS grib2 input data interpolating to the static data under fix.non.frac.grid/RRFS_CONUS_3km/ . A failure is successfully caused during the read of slope type from the static surface field and the desired output message is written to the log file.

Doxygen built with no warnings on WCOSS2 using 8608666.

## DEPENDENCIES:
N/A

## DOCUMENTATION:
N/A

## ISSUE: 
Addresses #906

### CODE REVIEW

**Comment by GeorgeGayno-NOAA** [comment]
And you tested this on your problem case? Can you note that under the 'tests conducted' section.

File: sorc/chgres_cube.fd/static_data.F90 [hunk 1]
```diff
@@ -289,6 +326,8 @@ subroutine read_static_file(field, i_target, j_target, tile, &
  real(esmf_kind_r8)                 :: num_days, num_days_rec1, rinc(5)
  real(esmf_kind_r8)                 :: weight_rec1, weight_rec2
```
Comment (GeorgeGayno-NOAA) [inline sorc/chgres_cube.fd/static_data.F90]: I don't think variable `land_frac_target_tile` is used.

File: sorc/chgres_cube.fd/static_data.F90 [hunk 2]
```diff
@@ -259,22 +295,23 @@ end subroutine get_static_fields
 !! yearly value on the tile
 !! @author George Gayno NCEP/EMC   
  subroutine read_static_file(field, i_target, j_target, tile, &
-                             data_one_tile, max_data_one_tile, &
+                             data_one_tile, land_frac, max_data_one_tile, &
                              min_data_one_tile)
  
  use netcdf
```
Comment (GeorgeGayno-NOAA) [inline sorc/chgres_cube.fd/static_data.F90:302]: I don't think `land_frac_target_grid` is used.

File: sorc/chgres_cube.fd/static_data.F90 [hunk 3]
```diff
@@ -259,22 +295,23 @@ end subroutine get_static_fields
 !! yearly value on the tile
 !! @author George Gayno NCEP/EMC   
  subroutine read_static_file(field, i_target, j_target, tile, &
```
Comment (GeorgeGayno-NOAA) [inline sorc/chgres_cube.fd/static_data.F90:298]: Add argument `land_frac` to the doxygen.

**Review by GeorgeGayno-NOAA — CHANGES_REQUESTED** [review]

**Comment by GeorgeGayno-NOAA** [comment]
@LarissaReames-NOAA - I just have three minor comments. Once you address them I can merge.

File: sorc/chgres_cube.fd/static_data.F90 (hunk 3 above)
Comment (LarissaReames) [reply sorc/chgres_cube.fd/static_data.F90:298]: Done in coming commit.

File: sorc/chgres_cube.fd/static_data.F90 (hunk 2 above)
Comment (LarissaReames) [reply sorc/chgres_cube.fd/static_data.F90:302]: Removed in coming commit.

File: sorc/chgres_cube.fd/static_data.F90 (hunk 1 above)
Comment (LarissaReames) [reply sorc/chgres_cube.fd/static_data.F90]: Removed in coming commit.

**Comment by GeorgeGayno-NOAA** [comment]
Please merge the latest updates from develop. Then, I will approve and merge.

**Comment by GeorgeGayno-NOAA** [comment]
Doxygen now builds without error using 5c99fb6.

**Comment by LarissaReames** [comment]
> Doxygen now builds without error using [5c99fb6](https://github.com/ufs-community/UFS_UTILS/commit/5c99fb691a7915d094db715dbde9b92eba98b85f).

Done.

**Review by GeorgeGayno-NOAA — APPROVED** [review]
Looks good. Will merge.

**Comment by GeorgeGayno-NOAA** [comment]
@LarissaReames-NOAA and @BenjaminBlake-NOAA Is this fix required for RRFSv1? If so, I will need to update the tag: https://github.com/ufs-community/UFS_UTILS/releases/tag/ufs_utils_1_12_2

**Comment by BenjaminBlake-NOAA** [comment]
@GeorgeGayno-NOAA  Since this issue came about after PR #873 was merged, I don't think so.  The tag for RRFSv1 was created just before the changes associated with 873.

**Pull request merged**